### PR TITLE
fixed DIG bugs for GS : 1814, 1819

### PR DESCRIPTION
**Pull request by minsunny**
https://github.com/nanoPayinc/NANOPAY/issues/1814
https://github.com/nanoPayinc/NANOPAY/issues/1819

fixed NPE errors

### CODE REVIEW

File: src/foam/nanos/dig/DigWebAgent.java [hunk 1]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
-          outputterJson.setOutputClassNames(false);
-          outputterJson.output(sink.getArray().toArray());
-
-          //resp.setContentType("application/json");
-          if ( emailSet ) {
-            output(x, outputterJson.toString());
-          } else {
-            out.println(outputterJson.toString());
+        if ( sink != null ) {
+          if (sink.getArray().size() == 0) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: Fix spacing of this line an next.

File: src/foam/nanos/dig/DigWebAgent.java [hunk 2]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
+            }
+            out.println("[]");
+            resp.setStatus(HttpServletResponse.SC_OK);
+            return;
           }
-        } else if ( Format.XML == format ) {
-          XMLSupport xmlSupport = new XMLSupport();
+          logger.debug(this.getClass().getSimpleName(), "objects selected: " + sink.getArray().size());
+
+          if (Format.JSON == format) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: spacing

File: src/foam/nanos/dig/DigWebAgent.java [hunk 3]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
+            foam.lib.json.Outputter outputterJson = new foam.lib.json.Outputter(OutputterMode.NETWORK);
+            outputterJson.setOutputDefaultValues(true);
+            outputterJson.setOutputClassNames(false);
+            outputterJson.output(sink.getArray().toArray());
+
+            //resp.setContentType("application/json");
+            if (emailSet) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: spacing:
if ( emailSet ) {

File: src/foam/nanos/dig/DigWebAgent.java [hunk 4]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
+              output(x, outputterJson.toString());
+            } else {
+              out.println(outputterJson.toString());
+            }
+          } else if (Format.XML == format) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: spacing

File: src/foam/nanos/dig/DigWebAgent.java [hunk 5]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
+          } else if (Format.CSV == format) {
+            foam.lib.csv.Outputter outputterCsv = new foam.lib.csv.Outputter(OutputterMode.NETWORK);
+            outputterCsv.output(sink.getArray().toArray());
 
-          List a = sink.getArray();
-          for ( int i = 0 ; i < a.size() ; i++ ) {
-            outputterCsv.put((FObject) a.get(i), null);
-          }
+            List a = sink.getArray();
+            for (int i = 0; i < a.size(); i++) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: spacing

File: src/foam/nanos/dig/DigWebAgent.java [hunk 6]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
-
-          outputterHtml.outputStartHtml();
-          outputterHtml.outputStartTable();
-          List a = sink.getArray();
-          for ( int i = 0 ; i < a.size() ; i++ ) {
-            if ( i == 0 ) {
-              outputterHtml.outputHead((FObject) a.get(i));
+            //resp.setContentType("text/plain");
+            //if ( email.length != 0 && ! email[0].equals("")  && email[0] != null ) {
+            if (emailSet) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: spacing

File: src/foam/nanos/dig/DigWebAgent.java [hunk 7]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
+              output(x, outputterCsv.toString());
+            } else {
+              out.println(outputterCsv.toString());
             }
-            outputterHtml.put((FObject) a.get(i), null);
-          }
-          outputterHtml.outputEndTable();
-          outputterHtml.outputEndHtml();
+          } else if (Format.HTML == format) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: spacing

File: src/foam/nanos/dig/DigWebAgent.java [hunk 8]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
+            foam.lib.html.Outputter outputterHtml = new foam.lib.html.Outputter(OutputterMode.NETWORK);
+
+            outputterHtml.outputStartHtml();
+            outputterHtml.outputStartTable();
+            List a = sink.getArray();
+            for (int i = 0; i < a.size(); i++) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: spacing this line and next

File: src/foam/nanos/dig/DigWebAgent.java [hunk 9]
```diff
@@ -231,91 +231,104 @@ public void execute(X x) {
...
-        }  else if ( Format.JSONJ == format) {
-          foam.lib.json.Outputter outputterJson = new foam.lib.json.Outputter(OutputterMode.NETWORK);
-          List a = sink.getArray();
-          String dataToString = "";
+            if (emailSet) {
+              output(x, outputterHtml.toString());
+            } else {
+              out.println(outputterHtml.toString());
+            }
+          } else if (Format.JSONJ == format) {
```
Comment (kgrgreer) [inline src/foam/nanos/dig/DigWebAgent.java]: spacing here and several places below

File: src/foam/nanos/dig/DigWebAgent.java (hunk 2 above)
Comment (minsunny) [reply src/foam/nanos/dig/DigWebAgent.java]: done

File: src/foam/nanos/dig/DigWebAgent.java (hunk 3 above)
Comment (minsunny) [reply src/foam/nanos/dig/DigWebAgent.java]: done

**Pull request merged**